fsspec/tests/test_spec.py

```diff
@@ -912,6 +984,20 @@ def test_dict_idempotent():
     assert DummyTestFS.from_dict(outa) is a
 
 
+def test_dict_json_serializable():
+    a = DummyTestFS(1)
+    b = DummyTestFS(2, bar=Path("baz"))
+    c = DummyTestFS(3, baz=b)
+
+    outa = a.to_dict()
+    outb = b.to_dict()
+    outc = c.to_dict()
+
+    json.dumps(outa)
+    json.dumps(outb)
+    json.dumps(outc)
```

Comment:
These show there is no error, but should we be checking their output too?

Reply:
The output is already checked by another test (`test_dict_fs_attr`). Here we specifically check whether the dict is serializable.